### PR TITLE
Add piping of arguments

### LINKED ISSUE #242: feat: Stream the directory entries
Currently eza buffers reads the entire directory and outputs the entire result at once. This is especially bad when using --tree

### PR DESCRIPTION
**Pull request by MartinFillon**
This pr is aiming at making available argument streaming

### CODE REVIEW

**Comment by MartinFillon** [comment]
Co-authored with @PThorpe92

File: src/options/mod.rs [hunk 1]
```diff
@@ -117,14 +120,17 @@ pub struct Options {
 
     /// The options to make up the styles of the UI and file names.
     pub theme: ThemeOptions,
+
+    /// Whether to read file names from stdin instead of the command-line
+    pub stdin: FilesInput,
 }
 
-impl Options {
+impl<'args> Options {
```
Comment (PThorpe92) [inline src/options/mod.rs:128]: Probably my fault but I believe this can be removed. 

I'll actually probably fix this (whole block I think) after work

Comment (MartinFillon) [reply src/options/mod.rs:128]: Ok I let you do that take your time

**Review by cafkafk — APPROVED** [review]
This is cool, and seems ready enough to merge, you can always follow up with fixes to lifetimes later.

**Pull request merged**